fission-web-client/library/Fission/Web/Client/V2.hs

```diff
@@ -0,0 +1,95 @@
+  ) where
+
+import           Servant.Client.Generic
+
+import           Fission.Web.API.Prelude
+
+import qualified Fission.Web.API.App.Types                        as App
+import qualified Fission.Web.API.Auth.UCAN.Types                  as Auth.UCAN
+import qualified Fission.Web.API.IPFS.Peer.Types                  as IPFS.Peer
+import qualified Fission.Web.API.IPFS.Types                       as IPFS
+import qualified Fission.Web.API.Types                            as Fission
+
+import qualified Fission.Web.API.User.DID.Types                   as User.DID
+import qualified Fission.Web.API.User.DataRoot.Types              as User.DataRoot
+import qualified Fission.Web.API.User.Email.Types                 as User.Email
+import qualified Fission.Web.API.User.Types                       as User
+import qualified Fission.Web.API.User.WhoAmI.Types                as User.WhoAmI
+
+import           Fission.Web.Client.Internal.Orphanage.Blaze.HTML ()
+import           Fission.Web.Client.Internal.Orphanage.WebSocket  ()
+
+Fission.Routes
+  { v2 = fromServant @_ @(AsClientT ClientM)->
+      Fission.RoutesV2
+        { api = fromServant @_ @(AsClientT ClientM) ->
+            Fission.V2
+              { app = fromServant @_ @(AsClientT ClientM) ->
+                  App.Routes
+                    { index   = appIndex
+                    , create  = createApp
+                    , update  = updateApp
+                    , destroy = destroyApp
+                    }
+
+              , auth = fromServant @_ @(AsClientT ClientM) ->
+                  Auth.UCAN.Routes {verify = verifyUCAN}
+
+              , ipfs = fromServant @_ @(AsClientT ClientM) ->
+                  IPFS.RoutesV2
+                    { peers = fromServant @_ @(AsClientT ClientM) ->
+                        IPFS.Peer.Routes {index = getIPFSPeers}
+                    }
+
+            , user = fromServant @_ @(AsClientT ClientM) ->
+                User.RoutesV2
+                  { create   = createUser
+                  , dataRoot = fromServant @_ @(AsClientT ClientM) ->
+                      User.DataRoot.Routes
+                        { get    = getDataRoot
+                        , update = setDataRoot
+                        }
+
+                  , whoAmI = fromServant @_ @(AsClientT ClientM) ->
+                      User.WhoAmI.Routes { whoAmI }
+
+                  , did = fromServant @_ @(AsClientT ClientM) ->
+                      User.DID.Routes
+                        { setAuthenticated = setDIDViaUCAN
+                        , setViaChallenge  = setDIDViaChallenge
+                        }
+
+                  , email = fromServant @_ @(AsClientT ClientM) ->
+                      User.Email.Routes
+                        { verify  = verifyViaEmail
+                        , resend  = resendVerificationEmail
+                        , recover = recoverViaEmail
+                        }
+                  }
+            }
+        }
+  } = genericClient
```

Comment:
That wins the price for the biggest pattern match I have ever seen LOL

There's some magic going on here. View patterns, i.e. `->` in pattern matches, right? And there's something happening with `fromServant`?

And in the end the values that are exposed are on the right hand side in the records, right? So e.g. `setDIDViaUCAN` or `resendVerificationEmail` are then the identifiers for the pattern matched pieces, right?

Just somewhat unfamiliar syntax for me right now :)

Reply:
Hey but, eventually this will all go away right? Because dot syntax. So it'll become `client.v2.api.user.email.resend` instead of `resendVerificationEmail`, right?

I've used up my quote for "right?" today. :P

Reply:
Yup, exactly! Can't wait for GHC 9.2

Reply:
> There's some magic going on here. View patterns, i.e. -> in pattern matches, right? And there's something happening with fromServant?

Yeah, view pattern magic here, just for convenience. I saw it in a post somewhere and liked how compact it is, but you're right that it requires explanation!

Inside each field is the standard `:<|>` version. `fromServant` takes that and converts it to the record style (there's also a `toServant` for going the other way). We can do that as a view pattern rather than pulling each of them out and converting them in separated functions.

Reply:
> That wins the price for the biggest pattern match I have ever seen LOL

😆 Right?!